[EN] Yohane (#24)
osu-datenshi/website-docs: en/yohane.md

title: "Yohane Commands"
old_id: 4
---
These are the commands for the Yohane bot.

## Yohane In-Game

### General commands

> - `!roll [number]` - Returns random number

Reviewer: ```suggestion
- `!roll [number]` - Roll a random set of number
```

Author: "set of number" is grammatically incorrect, and as it only returns one number, referring to it as "set" would be wrong

Author: I would also suggest to change the argument from "number" with eg "max" (I think it's the maximum amount? It's not really clear based on the given description)

Reviewer: Kind of confused on how to word it. I will leave it up to you 🤔

Author: I already did updates on the last commit, feel free to check

Reviewer: Checked, it's good 😎

- `!help` - Shows a list of all commands
- `!subscribe pp [options]` - Notifies about the pp status such as limit after [submitting a score](https://cdn.discordapp.com/attachments/265909019976138754/801454622346444860/unknown.png). (options: `set`,`get`,`unset`,`toggle`)
- `!boardshow [options]` - [Experimental] Hide yourself from the leaderboard (options: `others`,`normal`)

> - `!report` - If you find a cheater, please use this command.

Reviewer 2: ```suggestion
- `!report` - If you found a cheater or annoying players, use this command for report to the staff.
```

Reviewer: ```suggestion
- `!report` - If you encounter any cheater or an annoying player, use this command to report it to the staff.
```

Reviewer: atau gini
```suggestion
- `!report` - If you encounter any cheater or an annoying player, use this command to notify our staff.
```

Author: You wouldn't add "annoying" in the sense of a report in a English doc, as it is too subjective

Reviewer: Ah, that's fair, any suggestion on what word to use?
Maybe misbehaving?

Author: Well, we could describe it generally as people who break the rules or something like that

Reviewer: Ah, good idea. Maybe something like this, up to you.
```suggestion
- `!report` - If you encounter any player breaking rules, use this command to notify our staff.
```

- `!faq list` - Returns a list of things that are often needed in chat.

### Multiplayer commands

> These are general commands for the use in multiplayer rooms. Please use `!mp help` to show more commands.

Reviewer: ```suggestion
Commands that are shown here are for general use. Please use `!mp help` to show more commands.
```
translasi multiplayer nya dari mana?

Reviewer 2: > translasi multiplayer nya dari mana?

belum di translate

Reviewer: oh woop
s gw salah liat tadi

Author: Very "wonky" formulation. Will rephrase it for the sake of correct English. Also, please keep comments in English.

Author: Actually, the description is kinda redundant with the title and the list itself (as we mention the help twice in a row), we can just leave that out entirely imo.

Reviewer: We could, since the title itself in self-explanatory

Author: Alright, will cut it in the next commit.

- `!mp help` - Shows a list of all multiplayer commands.
- `!mp make [name]` - Creates a multiplayer room.
- `!mp close` - Closes the current multiplayer room.

> - `!mp checkperm` - Returns the permission level of the sender.

Reviewer: ```suggestion
- `!mp checkperm` - Display your permission level for the current room. 
```

- `!mp listref` - Displays a list of referees in the room.
- `!mp addref <username>` - Adds user as a referee in the room.
- `!mp rmref <username>` - Removes user as referee in the room.

> - `!mp start [timer]` - Sets all players to ready automatically and starts the game after a timeout (Default: `0`). Can be used to play solo in a multiplayer room.

Reviewer: ```suggestion
- `!mp start [timer]` - Automatically sets all players to be ready and starts the game after a countdown (Default: `0`). Can be used to play solo in a multiplayer room.
```

- `!mp abort` - Aborts the game. *This is different from using the `Escape`-key in solo*

> Note: For every operation using the username lookup, you can use `#<UserID>` instead.

Reviewer: ```suggestion
Note: For every operation that use Username Lookup, you can use `#<UserID>` instead.
```

Author: This is not English

Reviewer: Does it need a quotation mark for the command Username Lookup? like
```suggestion
Note: For every operation that uses 'Username Lookup' can use `#<UserID>` instead.

Author: We can do that, but I wouldn't say it's necessary. Whatever you prefer.

Reviewer: Ah, the original file in ID doesn't have quotation, but it does capitalize the commands, so maybe just this.
```suggestion
Note: For every operation using the Username Lookup, you can use `#<UserID>` instead.
```

Author: it is already changed in the latest commit

Reviewer: Oh sorry, nevermind then

> - `!fokabot reconnect` - Reconnect YohaneBOT if he's not on online users list anymore  

Reviewer: Yoooo BOT nya laki? he???

Reviewer 2: change to `!bot reconnect` also `"she"` because that is from ripple bot

Author: I will also take the opportunity to correct the sentence

    
    | Parameter | Value | Function |
    | :-- | :-: | :-- |
    | `mode` | `vanilla`, `relax` | Specifies the game mode for the *whitelist* |
    | `type` | `total`, `all` | Specifies the type of limit for the *whitelist* |
    | `procedure` | `add`, `set` | *Optional*<br>Specifies the operation of changing the PP value for the *whitelist* |
- `!map <command> <type> <ID>` - Performs operations on a map(set)
    
    | Parameter | Value | Function |
    | :-- | :-: | :-- |
    | `command` | `rank`, `love`, `unrank`, `reset` | Directly changes the status of the map. |
    | `command`\* | `auto-rank`, `auto-love`, `auto-cancel` | **Can only be set by someone permitted to manage autoranks**<br>Sets the eligibility of a map to get auto-rank. |
    | `type` | `set`, `set-of`, `map` | `set` - uses **mapset ID** to retrieve data.<br>`set-of` - uses **map ID** to get **a list of maps with the mapset ID** from the given **map ID**.<br>`map` - uses **map ID** to retrive data. |
    
    For more details, you may take a look at [this example](https://cdn.discordapp.com/attachments/265909019976138754/801471868699410473/unknown.png).

### Note

> If you want to use commands through Yohane's private message, you don't need to use the "!"-prefix.

Reviewer: ```suggestion
If you're using commands through Yohane's private message, you can type your command without the "!"-prefix.
```

Author: word doubling with "command", I'd only change the first part

Reviewer: could remove the latter

- Outside of that, you need to use the "!"-prefix to invoke Yohane's commands.

## YohaneBot Discord
- `@faq help` - Shows all commands

> - `@stats [options] [nickname]` - Displays profile status (options: `vanilla`,`relax`)

Reviewer 2: ```suggestion
- `@stats [options] [mode]` - Display profile status (options: `vanilla`, `relax`) (mode: `std`, `taiko`, `mania`, `ctb`)

> - `@recent [options] [nickname]` - Displays the most recent score (options: `vanilla`,`relax`)

Reviewer 2: ```suggestion
- `@recent [options]` - Displays the most recent score (options: `vanilla`, `relax`)
```

> - `@top [options] [nickname]` - Displays the top score (options: `vanilla`,`relax`)

Reviewer 2: ```suggestion
- `@top [options]` - Displays the top score (options: `vanilla`, `relax`).
```
